src/drivers/camera_capture/camera_capture.cpp

```diff
@@ -0,0 +1,313 @@
+ *
+ *   Copyright (c) 2017 PX4 Development Team. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in
+ *    the documentation and/or other materials provided with the
+ *    distribution.
+ * 3. Neither the name PX4 nor the names of its contributors may be
+ *    used to endorse or promote products derived from this software
+ *    without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+
+/**
+ * @file camera_capture.cpp
+ *
+ * Online and offline geotagging from camera feedback
+ *
+ * @author Mohammed Kabir <kabir@uasys.io>
+ */
+
+#include "camera_capture.hpp"
+
+#define commandParamToInt(n) static_cast<int>(n >= 0 ? n + 0.5f : n - 0.5f)
+
+namespace camera_capture
+{
+CameraCapture	*g_camera_capture;
+}
+
+CameraCapture::CameraCapture() :
+	_capture_enabled(false),
+	_trigger_pub(nullptr),
+	_command_ack_pub(nullptr),
+	_command_sub(-1),
+	_camera_capture_feedback(false),
+	_camera_capture_mode(0),
+	_camera_capture_edge(0),
+	_capture_seq(0),
+	_last_fall_time(0),
+	_last_exposure_time(0),
+	_capture_overflows(0)
+{
+
+	memset(&_work, 0, sizeof(_work));
+
+	// Parameters
+	_p_strobe_delay = param_find("CAM_CAP_DELAY");
+	param_get(_p_strobe_delay, &_strobe_delay);
+
+	_p_camera_capture_feedback = param_find("CAM_CAP_FBACK");
+	param_get(_p_camera_capture_feedback, &_camera_capture_feedback);
+
+	_p_camera_capture_mode = param_find("CAM_CAP_MODE");
+	param_get(_p_camera_capture_mode, &_camera_capture_mode);
+
+	_p_camera_capture_edge = param_find("CAM_CAP_EDGE");
+	param_get(_p_camera_capture_edge, &_camera_capture_edge);
+
+	if (_camera_capture_feedback) {
+		struct camera_trigger_s trigger = {};
```

Comment:
This is publishing a null message. Is that of consequence? Once this get fixed in uOrb this can use the new API. Until then using advertise on first publication (not on an ISR) will work.